commit c2d79d6b85f5804842f5a4dd55b643c2a8ec32b6
Author: Sakari Ailus <sakari.ailus@linux.intel.com>
Date:   Wed Jul 20 08:32:49 2016 -0300

    media: entity: Fix stream count check
    
    commit 41387a59c8fd55975c6a26cc12fc5c9ca61fcc0f upstream.
    
    There's a sanity check for the stream count remaining positive or zero on
    error path, but instead of performing the check on the traversed entity it
    is performed on the entity where traversal ends. Fix this.
    
    Fixes: commit 3801bc7d1b8d ("[media] media: Media Controller fix to not let stream_count go negative")
    
    Signed-off-by: Sakari Ailus <sakari.ailus@linux.intel.com>
    Reviewed-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
    Signed-off-by: Mauro Carvalho Chehab <mchehab@s-opensource.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/media/media-entity.c b/drivers/media/media-entity.c
index c68239e..98b067b 100644
--- a/drivers/media/media-entity.c
+++ b/drivers/media/media-entity.c
@@ -468,7 +468,7 @@ __must_check int __media_entity_pipeline_start(struct media_entity *entity,
 
 	while ((entity_err = media_entity_graph_walk_next(graph))) {
 		/* don't let the stream_count go negative */
-		if (entity->stream_count > 0) {
+		if (entity_err->stream_count > 0) {
 			entity_err->stream_count--;
 			if (entity_err->stream_count == 0)
 				entity_err->pipe = NULL;
